### PR TITLE
Add more log ACQFs

### PR DESCRIPTION
**Pull request by jduerholt**
This PR adds the following things:
- qLogNEI as new default ACQF for SOBO
- new MOBO strategy which features all functionality of the QnehviStrategy and the QehviStrategy. These strategies are now deprecated and will be removed in a future release. 
- MOBO features the following ACQFs:
    - qLogNEHVI (default one)
    - qNEHVI
    - qEHVI
    - qLogEHVI
 - Botorch 0.9.4 as minimal required version

### CODE REVIEW

File: tests/bofire/strategies/test_mobo.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        for acqf in [
+            acquisitions.qEHVI,
+            acquisitions.qLogEHVI,
+            acquisitions.qNEHVI,
+            acquisitions.qLogNEHVI,
+        ]
+        for num_test_candidates in range(1, 3)
+    ],
+)
+def test_mobo(strategy, use_ref_point, acqf, num_test_candidates):
```
Comment (simonsung06) [inline tests/bofire/strategies/test_mobo.py:90]: is num_test_candidates being used in this function?

**Review by simonsung06 — COMMENTED** [review]
Looks good. One minor thing and then I think it's ready. I like the decision to use the new Log-based acq functions!

File: tests/bofire/strategies/test_mobo.py (hunk 1 above)
Comment (jduerholt) [reply tests/bofire/strategies/test_mobo.py:90]: Good point, this is an copy paste thing, I will remove it.

Comment (jduerholt) [reply tests/bofire/strategies/test_mobo.py]: Fixed it.

**Review by simonsung06 — APPROVED** [review]
Thanks! Will merge it.

**Pull request merged**